src/pixbuf-util.cc

```diff
@@ -395,30 +395,23 @@ GdkPixbuf *pixbuf_fallback(FileData *fd, gint requested_width, gint requested_he
  */
 
 gboolean util_clip_region(gint x, gint y, gint w, gint h,
-			  gint clip_x, gint clip_y, gint clip_w, gint clip_h,
-			  gint *rx, gint *ry, gint *rw, gint *rh)
+                          gint clip_x, gint clip_y, gint clip_w, gint clip_h,
+                          gint &rx, gint &ry, gint &rw, gint &rh)
 {
-	// Ensures that clip region and main region have some overlap (they aren't
-	// completely disjoint).
-	if (clip_x + clip_w <= x ||  /* assert(x < clip_right) && */
-	    clip_x >= x + w ||       /* assert(clip_x < right) && */
-	    clip_y + clip_h <= y ||  /* assert(y < clip_bottom && */
-	    clip_y >= y + h)         /* assert(bottom < clip_y) */
+	GdkRectangle main{x, y, w, h};
+	GdkRectangle clip{clip_x, clip_y, clip_w, clip_h};
+	GdkRectangle r;
+
+	gboolean res = gdk_rectangle_intersect(&main, &clip, &r);
```

Comment:
Instead of `res`, I think it'd be more helpful to name the variable for what it actually represents.  So maybe `found_intersection`?  Also, as a habit, this can be `const`

Reply:
Renamed to `rectangles_intersect`.

Reply:
lgtm